### PR TITLE
*: add foreign key constraint check when execute update statement

### LINKED ISSUE #38155: support foreign key constrain check when execute update statement
## Dev Task

- [ ] Add foreign key constraint check when update foreign key child/parent table.

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #38155
 
Problem Summary: Add foreign key constraint check when execute update statement

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/foreign_key.go [hunk 1]
```diff
@@ -77,6 +77,40 @@ func (p *Insert) buildOnDuplicateUpdateColumns() map[string]struct{} {
...
+	if !ctx.GetSessionVars().ForeignKeyChecks {
+		return nil, nil
+	}
+	tblID2UpdateColumns := updt.buildTbl2UpdateColumns()
+	fkChecks := make(map[int64][]*FKCheck)
+	for tid, tbl := range tblID2table {
+		tblInfo := tbl.Meta()
+		dbInfo, exist := is.SchemaByTable(tblInfo)
+		if !exist {
+			return nil, infoschema.ErrDatabaseNotExists
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply planner/core/foreign_key.go:90]: normally, it should never happen, I will add a comment here.

[the remark this answers is not in the document]
Comment (crazycs520) [reply planner/core/foreign_key.go:91]: done.

**Review by winoros — APPROVED** [review]

File: planner/core/point_get_plan.go [hunk 2]
```diff
@@ -1546,6 +1546,11 @@ func buildPointUpdatePlan(ctx sessionctx.Context, pointPlan PhysicalPlan, dbName
 			updatePlan.PartitionedTable = append(updatePlan.PartitionedTable, pt)
 		}
 	}
+	fkChecks, err := updatePlan.buildOnUpdateFKChecks(ctx, is, updatePlan.tblID2Table)
+	if err != nil {
+		return nil
+	}
+	updatePlan.FKChecks = fkChecks
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply planner/core/point_get_plan.go]: ok, done.

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**